hub/core/transform/transform.py

```diff
@@ -152,13 +163,15 @@ def run_pipeline(
     if size_list:
         size_list[-1] -= extra
 
+    init_tensor_metas = {tensor: TensorMeta.load(tensor, storage) for tensor in tensors}
     all_workers_metas = compute.map(
         store_shard,
         zip(
             shards,
             size_list,
             repeat(storage),
             repeat(tensors),
```

Comment:
if we have tensor metas with keys containing tensor names, perhaps we should skip passing tensors?